### PR TITLE
soft delete video for a course

### PR DESCRIPTION
**Pull request by muhammad-ammar**
### Description

Add support to soft delete a video for a particular course.

### Testing
- [x] Unit

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review: @muzaffaryousaf 
- [x] Code review: @mushtaqak 

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: edxval/tests/test_api.py [hunk 1]
```diff
@@ -1055,3 +1055,47 @@ def test_invalid_encoded_video_attr(self):
     def test_invalid_course_id(self):
         xml = self.make_import_xml(video_dict=constants.VIDEO_DICT_FISH)
         self.assert_invalid_import(xml, "x" * 300)
+
+
+class GetCourseVideoRemoveTest(TestCase):
+    """
+    Tests for our remove_video_for_course function in api.py
```
Comment (mushtaqak) [inline edxval/tests/test_api.py]: I would have written `Tests to check 'remove_video_for_course' method works correctly.` or something like that.

File: edxval/tests/test_api.py [hunk 2]
```diff
@@ -1055,3 +1055,47 @@ def test_invalid_encoded_video_attr(self):
...
+    """
+
+    def setUp(self):
+        """
+        Creates Video objects in database
```
Comment (mushtaqak) [inline edxval/tests/test_api.py]: I would have written it as `Creates video objects for courses` or something like that.

**Comment by muhammad-ammar** [comment]
@mushtaqak Added your suggestions.

File: edxval/api.py [hunk 3]
```diff
@@ -328,12 +328,25 @@ def get_videos_for_course(
...
-        {"courses__course_id":unicode(course_id)},
+        {"courses__course_id": unicode(course_id), "courses__is_hidden": False},
         sort_field,
         sort_dir,
     )
 
 
+def remove_video_for_course(course_id, edx_video_id):
+    """
+    Remove video for a particular course.
```
Comment (muzaffaryousaf) [inline edxval/api.py]: `Soft deletes video for particular course.`

File: edxval/models.py [hunk 4]
```diff
@@ -125,6 +125,7 @@ class CourseVideo(models.Model, ModelFactoryWithValidation):
     """
     course_id = models.CharField(max_length=255)
     video = models.ForeignKey(Video, related_name='courses')
+    is_hidden = models.BooleanField(default=False, help_text='Hide this video for course.')
```
Comment (muzaffaryousaf) [inline edxval/models.py]: `Hide video for course.`

File: edxval/tests/test_api.py [hunk 5]
```diff
@@ -1055,3 +1055,47 @@ def test_invalid_encoded_video_attr(self):
...
+        """
+        Creates video objects for courses
+        """
+        # create video in the test course
+        self.course_id = 'test-course'
+        video = Video.objects.create(**constants.VIDEO_DICT_FISH)
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+        self.edx_video_id = video.edx_video_id
+
+        # add the video in another course (to make sure that video is remove for correct course )
```
Comment (muzaffaryousaf) [inline edxval/tests/test_api.py]: `remove` --> `removed`

**Comment by muzaffaryousaf** [comment]
@muhammad-ammar This LGTM 👍 except couple of suggestions.

**Comment by muzaffaryousaf** [comment]
Squash & merge.

**Comment by muhammad-ammar** [comment]
We are not handling the the exception in case video doesn't exist. So i am
not sure if we really need to write a test for this case.

**Comment by BenjiLee** [comment]
In that case, should we be handling that exception?

What happens when you try to delete a video that doesn't exist? 404? 500? 200?

**Comment by muhammad-ammar** [comment]
@BenjiLee We get `DoesNotExist` exception when video doesn't exist. In
edxval at other places we catch this exception and then raised another
exception which is `ValVideoNotFoundError`. But in video removal case i am
not raising `ValVideoNotFoundError`. We are using edxval as an internal api
from within edx-platform without any HTTP REST interface so there is no
HTTP status code and we will get an unhandled exception whether we raise
`DoesNotExist` or `ValVideoNotFoundError`.

We are deleting videos from studio `Video Uploads` page so this case(video
doesn't exist) will not happen. Even if for some reason this happens,
studio will show a default error popup.

Do you see any issue with current approach?

**Comment by BenjiLee** [comment]
Ah, didn't realize right away it was an internal call. Good to go.

**Pull request merged**